django/test/utils.py

```diff
@@ -841,6 +844,43 @@ def disable(self):
         setattr(Options, 'default_apps', self.old_apps)
 
 
+class time_keeper:
+    def __init__(self, n_tests):
+        self.test_suite_records = {}
+        self.records = []
+        self.n_tests = n_tests
+
+    def append(self, record, record_time):
+        if 'test' in record:
+            self.test_suite_records[record] = record_time
+        else:
+            new_record = '%s took %.3fs' % (record, record_time)
+            self.records.append(new_record)
+
+    @contextmanager
+    def timed(self, name):
+        start_time = time.perf_counter()
+        try:
+            yield
+        finally:
+            end_time = time.perf_counter() - start_time
+            self.append(name, end_time)
+
+    def results(self):
+        results = []
+        if self.n_tests > 0:
+            sorted_times = sorted(self.test_suite_records.items(), reverse=True, key=lambda records: records[1])
+            for key, value in sorted_times:
+                if self.n_tests > 0:
+                    results.append("%s took %0.3fs" % (key, value))
```

Comment:
I'm not sure about this string. It results in the time (the most important bit) being at different indentations. I'd prefer to see it at the start so it is easier to read the timings - the test name itself is of secondary importance. 

For reference, Pytest formats strings like this. 
`0.50s call test_this.py::SlowTestCase::test_should_run_slow`

Reply:
That's a good idea. I'll change it